plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -791,6 +792,9 @@ def __init__(self, **kwargs: Any):
             "gas_price_strategies", DEFAULT_GAS_PRICE_STRATEGIES
         )
 
+        if kwargs.pop("poa_chain", False):
+            self._api.middleware_onion.inject(geth_poa_middleware, layer=0)
+
```

Comment:
Then we configure the variable to true in our aea-config.yaml for polygon deployment.

Reply:
Depends. Did you test that it works without default value being set on config file?

Reply:
It'll work without updating the config file since we're using `False` as default if key not found. But I think you should update the connection configs on this repo.

Reply:
The problem isn't the dict here.. the problem is the aea config layer. It usually requires all keys to be set, undefined keys cannot be overriden.